### PR TITLE
Replace virtus objects with dry-struct objects

### PR DESCRIPTION
**Pull request by indrekj**
dry-struct has more flexible type interface.

### CODE REVIEW

**Comment by indrekj** [comment]
@urmastalimaa

File: spec/salestation/app/errors/dependency_currently_unavailable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe Salestation::App::Errors::DependencyCurrentlyUnavailable do
+  subject(:create_dependency_currently_unavailable) { described_class.new(message: message) }
+
+  let(:message) { 'message' }
+
+  it 'has message' do
+    expect(create_dependency_currently_unavailable.message).to eq(message)
```
Comment (urmastalimaa) [inline spec/salestation/app/errors/dependency_currently_unavailable_spec.rb]: The setup for this test feels a bit forced, I see where you are going with this, but perhaps just
`expect(described_class.new(message: message).message).to eq(message)`

File: spec/salestation/web/responses/success_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require 'spec_helper'
+
+describe Salestation::Web::Responses::Success do
+  subject(:create_success) { described_class.new(attributes) }
+  let(:attributes) { all_attributes }
```
Comment (urmastalimaa) [inline spec/salestation/web/responses/success_spec.rb]: Unnecessary indirection.

Comment (urmastalimaa) [inline spec/salestation/web/responses/success_spec.rb]: I thought about it for other specs, but couldn't come up with anything particularly good.
But for this test specifically you are not removing any attributes.

**Comment by willmore** [comment]
!squash

**Pull request merged**